### PR TITLE
Fix the search component exit behaviour.

### LINKED ISSUE #6414: Settings UI - Search: can't exit search mode when ⨉ is clicked
When you click the **⨉** to close the search, if the field is empty, you exit the search mode, as can be seen here
    cantdisable

### PR DESCRIPTION
**Pull request by zinigor**
Fixes #6414 

#### Changes proposed in this Pull Request:
* Updates tests to use the `before` functionality of the testing framework.
* Modifies the way search handles the query fragment - only sets it to `search` when there is actually a search term present.
* Fixes some eslint warnings.

This PR is waiting for https://github.com/Automattic/dops-components/pull/93 to get completed and merged.

### CODE REVIEW

**Comment by eliorivero** [comment]
1. got this warning as soon as I load the settings UI
```
Warning: Failed propType: Invalid prop `children` of type `array` supplied to `SelectDropdownItem`, expected `string`. Check the render method of `NavTabs`.
```

2. it currently looks like this
captura de pantalla 2017-02-20 a las 18 55 36

**Comment by zinigor** [comment]
Fixes #6416

**Comment by eliorivero** [comment]
If you're in `#/settings` (which is quite common, since when you click the Settings button in the button group or the menu link, you land at `#/settings`), when you search and click the x, you go to the Dashboard.

**Comment by zinigor** [comment]
Fixes #6415

**Comment by zinigor** [comment]
Thanks for reviewing @eliorivero ! I have refactored how search works by completely removing the Search route. Now we only use a search string.
Note: you need to update the companion dops-components PR to test this PR.

**Comment by zinigor** [comment]
Travis fails here because it needs a dops-component counterpart.

**Comment by samhotchkiss** [comment]
Which dops-component PR needs merging?

File: _inc/client/components/navigation-settings/index.jsx [hunk 1]
```diff
@@ -30,72 +28,60 @@ import { getSiteConnectionStatus } from 'state/connection';
...
-					onSearch={ this.onSearch }
-					onSearchClose={ this.onClose }
-					onBlur={ this.onBlur }
-					isOpen={
-						'/search' === this.props.route.path
-						|| this.props.searchHasFocus
-					}
+					onSearch={ this.doSearch }
+					isOpen={ false !== this.props.searchTerm }
+					initialValue={ this.props.searchTerm }
```
Comment (eliorivero) [inline _inc/client/components/navigation-settings/index.jsx]: I get this console warning when I load the page
```
Warning: Failed propType: Invalid prop `initialValue` of type `boolean` supplied to `Search`, expected `string`. Check the render method of `NavigationSettings`.
```
maybe this line?

I see that you're returning different types (boolean, string) in the reducer.

File: _inc/client/state/search/reducer.js [hunk 2]
```diff
@@ -10,9 +10,7 @@ import { combineReducers } from 'redux';
  * Internal dependencies
  */
 import {
-	JETPACK_SEARCH_TERM,
-	JETPACK_SEARCH_FOCUS,
-	JETPACK_SEARCH_BLUR
+	JETPACK_SEARCH_TERM
 } from 'state/action-types';
 
 const searchTerm = ( state = false, action ) => {
```
Comment (eliorivero) [inline _inc/client/state/search/reducer.js]: The initial state should be `''` instead of `false` to match types.

**Comment by zinigor** [comment]
@samhotchkiss the dops-components PR is this one: https://github.com/Automattic/dops-components/pull/93

@eliorivero thanks for the review! Your comments about style issues and the search component behaviour are better suited to the dops-component PR, I'll move them there.

**Comment by zinigor** [comment]
Thanks @eliorivero , I have changed the default search value to be a string, not a boolean. That fixes the  problem with search being open on reloading.

**Pull request merged**